### PR TITLE
unexport buggy function nodeSelectorAsSelector

### PR DESCRIPTION
**Pull request by aojea**
/kind bug
/kind cleanup

```release-note
NONE
```

just to avoid this gets consumed, it has to be fixed anyway

### CODE REVIEW

File: pkg/controller/nodeipam/ipam/multi_cidr_range_allocator.go [hunk 1]
```diff
@@ -1257,3 +1257,62 @@ func listClusterCIDRs(ctx context.Context, kubeClient clientset.Interface) (*net
...
+	default:
+		return nil, fmt.Errorf("%q is not a valid node selector operator", nsr.Operator)
+	}
+	return labels.NewRequirement(nsr.Key, op, nsr.Values)
+}
+
+// nodeSelectorAsSelector converts the NodeSelector api type into a struct that
+// implements labels.Selector
+// Note: This function should be kept in sync with the selector methods in
+// pkg/labels/selector.go
```
Comment (liggitt) [inline pkg/controller/nodeipam/ipam/multi_cidr_range_allocator.go:1287]: thanks, mind adding a TODO with a link to the issue to fix with a comment that the current impl is not correct

Comment (aojea) [reply pkg/controller/nodeipam/ipam/multi_cidr_range_allocator.go:1290]: added

**Pull request merged**